### PR TITLE
PDF annotations as notes

### PR DESCRIPTION
**Pull request by fnevgeny**
PR for issue #291. In addition:

1. pdfpc now recognizes more types of PDF annotations;
2. If there are few per page, they will be concatenated;
3. Editing PDF-imported notes is disabled which is indicated by a flashing "lock" icon if 'n' is pressed.

### CODE REVIEW

File: src/classes/metadata/pdf.vala [hunk 1]
```diff
@@ -363,17 +363,35 @@ namespace pdfpc.Metadata {
          * Fill the slide notes from pdf text annotations.
          */
         private void notes_from_document() {
-            for(int i = 0; i < this.page_count; i++) {
+            for (int i = 0; i < this.page_count; i++) {
                 var page = this.document.get_page(i);
+                int user_slide = real_slide_to_user_slide(i);
+                string note_text = this.notes.get_note_for_slide(user_slide);
```
Comment (AndreasBilke) [inline src/classes/metadata/pdf.vala]: I would make it clear, that `note_text` comes from the pdfpc file. Like
`this.notes.get_note_from_pdfpc_for_slide` and `this.notes.get_note_from_pdf_for_slide`

**Comment by AndreasBilke** [comment]
I tried it with a small example. Really nice!

File: src/classes/metadata/pdf.vala (hunk 1 above)
Comment (fnevgeny) [reply src/classes/metadata/pdf.vala]: Actually, this.notes contain both types of notes, and so get_note_for_slide may return either. There is a is_note_native() method to tell them apart. I was hesitating whether to alter get_note_for_slide() to return the entire slide_note structure instead...

**Comment by fnevgeny** [comment]
I realized I should had added a dedicated "lock" icon instead of re-using this.blank_icon. Since if the presentation is turned off ('b' pressed) and the user tries to edit a PDF annotation , flashing the same icon will clear the indication that the presentation view is disabled. Of course, the chances of this sequence of events is next to zero, it is still not nice...

**Pull request merged**